website/api/manager/ManagerAPI.php

```diff
@@ -81,20 +88,29 @@ protected function task() {
                 );
             }
 
+            // don't give out any task if games have been stopped.
+            if($this->checkConfig("compState", "noGameTasks")) {
+                return array("type" => "notask");
```

Comment:
Nope, it's written that way to stop them from throwing an exception.

It looks like the worker was intended to deal with a blank return correctly but it fails to do so and throws an exception. This lets them sleep for 2 seconds since they didn't get a task.